src/app/components/MediaLoader/configs/portraitClipMedia.ts

```diff
@@ -6,93 +6,47 @@ import {
   ConfigBuilderReturnProps,
   PlaylistItem,
 } from '../types';
-import shouldDisplayAds from '../utils/shouldDisplayAds';
-import AUDIO_UI_CONFIG from './constants';
 
 const DEFAULT_WIDTH = 512;
 
+export const setImageWidth = (url?: string) =>
+  url?.replace('{width}', String(DEFAULT_WIDTH));
+
 export default ({
   blocks,
   basePlayerConfig,
-  adsEnabled = false,
-  showAdsBasedOnLocation = false,
 }: ConfigBuilderProps): ConfigBuilderReturnProps => {
-  const firstBlock = filterForBlockType(
-    blocks,
-    'portraitClipMedia',
-  ) as PortraitClipMediaBlock;
-
-  if (!firstBlock) {
-    return {
-      mediaType: 'video',
-      playerConfig: basePlayerConfig,
-      showAds: false,
-      orientation: 'portrait',
-    };
-  }
+  const { model }: PortraitClipMediaBlock =
+    filterForBlockType(blocks, 'portraitClipMedia') ?? {};
 
-  const portraitClipMediaBlocks: PortraitClipMediaBlock[] = filterForBlockType(
-    blocks,
-    'portraitClipMedia',
-    { returnAllMatchingBlocks: true },
-  ) as PortraitClipMediaBlock[];
+  const { video, images = [] } = model || {};
 
-  const playlistItems: PlaylistItem[] = portraitClipMediaBlocks.map(block => {
-    const { model } = block;
-    const { video, images } = model;
-    const version = video?.version;
+  const { id, title, version } = video || {};
 
-    // prefer portrait-oriented image if available (based on our bff structure) fallback to first available
-    const image = images?.[1] || images?.[0];
-    const holdingImageURL = image?.urlTemplate?.replace(
-      '{width}',
-      `${DEFAULT_WIDTH}`,
-    );
+  const holdingImageURL = setImageWidth(
+    (images?.[1] || images?.[0])?.urlTemplate,
```

Comment:
Is there a nicer way to find the image with portrait orientation, or is it always the 2nd item in the list if it exists? Could we use destructring of the images array to make this more readable?

```
const [fallbackImage, portraitImage] = images;
```

```suggestion
    (portraitImage || fallbackImage)?.urlTemplate,
```

Reply:
Good shout on the destruct. One for @pvaliani perhaps. I _think_ the portrait version comes through second reliably?

Reply:
Yep agreed. The destructuring you suggested:

`const [fallbackImage, portraitImage] = images;` is already in place now, and we’re using `(portraitImage || fallbackImage)?.urlTemplate` to get the portrait-oriented image if it exists, otherwise falling back.

We’ve observed that the portrait image reliably comes second in the array (based on current `Home` page for Portuguese), so this fallback approach should ensure resilience in case that ever changes

thanks for reviewing @karinathomasbbc and @amoore108